components/bl-AnimateOnScroll-component/src/index.js

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useMemo, useRef } from 'react';
+import { Aos } from './aos';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const DevicesMap = {
+  mobile: 'mobile',
+  tablet: 'tablet',
+  all   : true,
+  none  : false,
+};
+
+export default function AnimateOnScroll({ component, eventHandlers, pods }) {
+  const {
+    display, classList, style, animationType, easing, side,
+    duration, offset, disableFor, delay, mirror, once, anchor,
+  } = component;
+  const { onAnimationEnter, onAnimationOut } = eventHandlers;
+  const elementPod = pods['animationPod'];
+
+  const animationOnScrollRef = useRef(null);
+
+  const animationName = useMemo(() => {
+    return animationType + (side === 'none' ? '' : `-${ side }`);
+  }, [animationType, side]);
+
+  useEffect(() => {
+    Aos.init({ disable: DevicesMap[disableFor] });
+
+    document.addEventListener('aos:in', () => onAnimationEnter());
```

Comment:
how does it work when on the page more than one instance?

Reply:
AOS dispatches two events on document: aos:in and aos:out whenever any element animates in or out

But I see the possibility of making events for the current element and not for any